modules/Microsoft.Storage/storageAccounts/deploy.bicep

```
@@ -378,7 +378,7 @@ module storageAccount_localUsers 'localUsers/deploy.bicep' = [for (localUser, in
     storageAccountName: storageAccount.name
     name: localUser.name
     hasSharedKey: contains(localUser, 'hasSharedKey') ? localUser.hasSharedKey : false
-    hasSshKey: contains(localUser, 'hasSshPassword') ? localUser.hasSshPassword : true
+    hasSshKey: contains(localUser, 'hasSshKey') ? localUser.hasSshKey : false
```

Good catch. It is however actually a 'required' value so it should be passed as follows:
```suggestion
    hasSshKey: localUser.hasSshKey
```

---

seems odd to only do that for the one parameter and not all of them, but ok.

---

Hey David,
that is true. I only commented on the line that was part of your PR. For the child-module deployment in general, all required parameter should be passed down as such:
- hasSshKey
- hasSshPassword
- name
- permissionScopes

Which means, the entire block should look like:
```bicep
module storageAccount_localUsers 'localUsers/deploy.bicep' = [for (localUser, index) in localUsers: {
  name: '${uniqueString(deployment().name, location)}-Storage-LocalUsers-${index}'
  params: {
    storageAccountName: storageAccount.name
    name: localUser.name
    hasSshKey: localUser.hasSshKey
    hasSshPassword: localUser.hasSshPassword
    permissionScopes: localUser.permissionScopes
    hasSharedKey: contains(localUser, 'hasSharedKey') ? localUser.hasSharedKey : false
    homeDirectory: contains(localUser, 'homeDirectory') ? localUser.homeDirectory : ''
    sshAuthorizedKeys: contains(localUser, 'sshAuthorizedKeys') ? localUser.sshAuthorizedKeys : []
    enableDefaultTelemetry: enableReferencedModulesTelemetry
  }
}]
```

However, why for example `hasSharedKey` isn't treated the same way, I don't know. The original contributor only considered the initial list as 'mandatory' values - which doesn't mean we could not change it.

---

Added, but we're maybe a bit out of scope of the original bugfix now... Can we get the CI to run and merge this now?

---

Hey @davidkarlsen as per the [Contribution Guide](https://github.com/Azure/ResourceModules/wiki/Contribution%20guide%20-%20Contribution%20flow) we would usually ask the contributor to setup the CI in their fork and provide a pipeline badge that shows that the contribution works. Unless folks don't have access to a subscription, that is.
This may change in the future and we may be able to enable users to run pipelines directly. But for the time being, we follow the flow description in the contribution guide.

That being said, the change is so small that it appears highly unlikely that it will fail. So we can skip that part.